File: thain-server/src/main/java/com/xiaomi/thain/server/model/dr/JobDr.java
package com.xiaomi.thain.server.model.dr;

import lombok.AllArgsConstructor;
import lombok.NonNull;

import java.sql.Timestamp;

/**
 * @author liangyongrui
 */
@AllArgsConstructor
public class JobDr {
    public final long id;
    public final long flowId;
    @NonNull
    public final String name;
    @NonNull
    public final String condition;
    @NonNull
    public final String component;
    @NonNull
    public final String callbackUrl;
    @NonNull
    public final String properties;
    public final int xAxis;
    public final int yAxis;
    @NonNull
    public final Timestamp createTime;
    public final boolean deleted;
}

File: thain-common/src/main/kotlin/com/xiaomi/thain/common/model/dr/JobDr.kt
package com.xiaomi.thain.common.model.dr

import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject
import java.sql.Timestamp

/**
 * Date 19-5-20 下午9:00
 *
 * @author liangyongrui@xiaomi.com
 */
class JobDr(val id: Long,
            val flowId: Long,
            val name: String,
            val condition: String,
            val component: String,
            val callbackUrl: String?,
            val properties: String,
            val xAxis: Int,
            val yAxis: Int,
            val createTime: Timestamp,
            val deleted: Boolean
) {
    val propertiesMap: JSONObject
        get() = JSON.parseObject(properties)
}
